crates/factor-outbound-networking/src/lib.rs

```diff
@@ -67,31 +67,36 @@ impl Factor for OutboundNetworkingFactor {
         .map(|res| res.map(Arc::new).map_err(Arc::new))
         .boxed()
         .shared();
-        // let prepared_resolver = resolver.prepare().await?;
-        // let allowed_hosts = AllowedHostsConfig::parse(
-        //         .context("missing component allowed hosts")?,
-        //     &prepared_resolver,
-        // )?;
-
-        // Update Wasi socket allowed ports
-        let wasi_preparer = builders.get_mut::<WasiFactor>()?;
-        let hosts_future = allowed_hosts_future.clone();
-        wasi_preparer.outbound_socket_addr_check(move |addr| {
-            let hosts_future = hosts_future.clone();
-            async move {
-                match hosts_future.await {
-                    Ok(allowed_hosts) => {
-                        // TODO: verify this actually works...
-                        spin_outbound_networking::check_url(&addr.to_string(), "*", &allowed_hosts)
-                    }
-                    Err(err) => {
-                        // TODO: should this trap (somehow)?
-                        tracing::error!(%err, "allowed_outbound_hosts variable resolution failed");
-                        false
+
+        match builders.get_mut::<WasiFactor>() {
+            Ok(wasi_builder) => {
+                // Update Wasi socket allowed ports
+                let hosts_future = allowed_hosts_future.clone();
+                wasi_builder.outbound_socket_addr_check(move |addr, addr_use| {
+                    let hosts_future = hosts_future.clone();
+                    async move {
+                        match hosts_future.await {
+                            Ok(allowed_hosts) => {
+                                // TODO: validate against existing spin-core behavior
+                                let scheme = match addr_use {
+                                    SocketAddrUse::TcpBind => return false,
+                                    SocketAddrUse::TcpConnect => "tcp",
+                                    SocketAddrUse::UdpBind | SocketAddrUse::UdpConnect | SocketAddrUse::UdpOutgoingDatagram => "udp",
+                                };
+                                spin_outbound_networking::check_url(&addr.to_string(),scheme, &allowed_hosts)
```

Comment:
This is a significant change from the way outbound sockets are checked today. It skips the address "Pool" and checks against the allowed hosts directly. This required adding an actual scheme for the checks, so it made sense to me to support "tcp" and "udp" schemes to allow filtering by those protocols.